packages/flutter_tools/lib/src/resident_runner.dart

```diff
@@ -1183,6 +1183,7 @@ abstract class ResidentRunner extends ResidentHandlers {
     Completer<void> appStartedCompleter,
     bool allowExistingDdsInstance = false,
     bool enableDevTools = false,
+    bool needsFullRestart = true,
```

Comment:
Can you add a doc comment on what `needsFullRestart` is for?

Reply:
Doc comments should go on the method itself. Look at the flutter framework for examples of how parameters are documented. Usually something like:

````
/// One sentence summary about what foo does
///
/// [bar] is blah blah.
void foo(String bar);
```

Reply:
I was following this particular example which apparently to have more than one line is acceptable. About going in the method agree I should move this up. https://dart.dev/guides/language/effective-dart/documentation#avoid-redundancy-with-the-surrounding-context